### DISCUSSION TITLE
1024kB RAM posible? unused bit d3 #dffd?

### DISCUSSION THREAD

**Discussion by tuFFo126** [UzixLS/zx-sizif-512#58] [General]
1024kB RAM.  If I insert a 1MB RAM stick into sizif, will sizif address another 512kB with the d3 bit of the #dffd port, because this bit is not used, because it would point to the void.  The same question applies to the GeneralSound sound extension, which normally has 512kB, will inserting a 1MB stick there make it visible?  I have mod sound files that are larger than 512kB, they load, GS plays them, but you can hear that not everything fits.

**Comment by UzixLS** [comment]
It's possible, but you need to modify CPLD firmware to get it working.

**Reply by tuFFo126** [reply]
The question is whether there is any software for Profi that uses more than 512kB, unless it's your own program that would save images and then make a stop-motion animation.  For more than 256kB for Profi I only found the badapple demo.  In the case of GeneralSound, it would be good to increase this memory for MOD files larger than 512kB.  For sizif there could be two firmwares to use, version 512, and version 1024.

**Reply by UzixLS** [reply]
Personally I'm using #dffd port caps for WildPlayer and NSID player, not sure about any other useful software for extra ram.

**Reply by tuFFo126** [reply]
I found some software for pentagon 512 or 1024 but it is not Profi, i.e. the memory above is supported by port #7ffd, the oldest bits not used in 128/+2.  so these programs cannot be run on sizif.  I wonder if it is possible to introduce an additional Machine mode: the current "Pentagon" mode would be called "Profi" with memory support >128 from the #dffd port, and a new "Pentagon" mode with memory support >128 through port #7ffd on bits d7, d6  , d5 (and here the meaning of d5 changes compared to the usual 128/+2) .  From what I read about Pentagons, not Profi, only memory above 512 was supported by the #dffd port. There would be two Pentagon standards.  is it possible to put it in cpld?

**Reply by UzixLS** [reply]
Can you share software you found? If this is decent list, I would to consider implementing an additional memory mode, but this is not what I would like. Extra bits in #7ffd port will cause incompatibilities with some random old software, which isn't case with #dffd.

**Reply by tuFFo126** [reply]
It would be possible to create a "Profi" mode, i.e. the current "Pentagon" with the #dffd port for >128.  And a new "Pentagon" mode with port #7ffd with bits d5, d6, d7 for addressing up to 512k and optional #dffd for memory above 512kB up to 1024kB.  Then there would be no compatibility problem because we would choose the right mode.  And I will pack the programs I have and share them. If the new firmware supported Pentagon mode with the #dffd port addressing memory above 512, the RAM would have to be replaced with an appropriate one to use it.  However, if we stayed at 512kB, we would have two ways of addressing memory above 128kB depending on the mode selected. 
Here's what I tried to run. I know, normal software 1024 dont run on 512. But thought, it woulding work,those software on 512 or 256 but not those:
https://zxaaa.net/view_demo.php?id=6441 (pentagon, atm) 512k
https://zxaaa.net/view_demo.php?id=9431 (pentagon) 512k
https://zxaaa.net/view_demo.php?id=8600 (pentagon) 256k
https://zxaaa.net/view_demo.php?id=8598 (pentagon) 256k
https://zxaaa.net/view_demo.php?id=7488 (pentagon) 256k
https://zxaaa.net/view_demo.php?id=7488 (pentagon) 512k ok z divIDE
https://zxaaa.net/view_demo.php?id=5038 (pentagon) 1024k
https://zxaaa.net/view_demo.php?id=6548 (profi) 1024k
https://zxaaa.net/view_demo.php?id=1820 (pentagon) 1024k
https://zxaaa.net/view_demo.php?id=1000 (pentagon) 1024k
https://zxart.ee/eng/software/demoscene/cache-file-manager-512k/
https://zxart.ee/eng/software/demoscene/gift/dismaster-gift-512k/

**Reply by tuFFo126** [reply]
Coming back to the question about 1024kB RAM for GeneralSound.  Because there are a lot of MOD files over 512kB.  Is it possible to handle larger memory, similarly to sizif itself?
![20240405_234552](https://github.com/UzixLS/zx-sizif-512/assets/124886433/85bb9ed2-c2de-46e2-aab0-3d3d60374a5c)
MOD files larger than 512kB are also loaded and played, but there are errors because not everything fits in memory.
